File: CS/MachineLearningDemo/Data/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ml.Attributes;
using System.ComponentModel;

namespace MachineLearningDemo.Data
{
    public enum Grade
    {
        A,
        B,
        C,
        D,
        F
    }

    public class Student
    {
        [Feature]
        public double GradeToPercent()
        {
            double grade = 0;
            switch (Grade)
            {
                case Grade.A:
                    grade = 90;
                    break;
                case Grade.B:
                    grade = 80;
                    break;
                case Grade.C:
                    grade = 70;
                    break;
                case Grade.D:
                    grade = 60;
                    break;
                case Grade.F:
                    grade = 50;
                    break;
            }
            return grade;
        }

        [Feature]
        public double QuadrupleFriends()
        {
            return Friends * 4;
        }

        public string GetName()
        {
            return Name;
        }

        public string Name { get; set; }

        public Grade Grade { get; set; }

        [Feature]
        public double GPA { get; set; }

        [Feature]
        public int Age { get; set; }

        [Feature]
        public bool Tall { get; set; }

        public int Friends { get; set; }

        [Label]
        public bool Good { get; set; }

        public int Cluster { get; set; }        
    }
}

File: VB/MachineLearningDemo/Data/Student.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports ml.Attributes
Imports System.ComponentModel

Namespace MachineLearningDemo.Data
    Public Enum Grade
        A
        B
        C
        D
        F
    End Enum

    Public Class Student
        <Feature> _
        Public Function GradeToPercent() As Double

            Dim grade_Renamed As Double = 0
            Select Case Grade
                Case Grade.A
                    grade_Renamed = 90
                Case Grade.B
                    grade_Renamed = 80
                Case Grade.C
                    grade_Renamed = 70
                Case Grade.D
                    grade_Renamed = 60
                Case Grade.F
                    grade_Renamed = 50
            End Select
            Return grade_Renamed
        End Function

        <Feature> _
        Public Function QuadrupleFriends() As Double
            Return Friends * 4
        End Function

        Public Function GetName() As String
            Return Name
        End Function

        Public Property Name() As String

        Public Property Grade() As Grade

        <Feature> _
        Public Property GPA() As Double

        <Feature> _
        Public Property Age() As Integer

        <Feature> _
        Public Property Tall() As Boolean

        Public Property Friends() As Integer

        <Label> _
        Public Property Good() As Boolean

        Public Property Cluster() As Integer
    End Class
End Namespace
